.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,15 @@
+name: pre-commit
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+    - uses: pre-commit/action@v2.0.3
```

Comment:
Apparently this action is [deprecated](https://github.com/pre-commit/action#readme). Might not be a problem, but are there any alternatives?

Reply:
deprecated action is not an issue, they try to push everybody to their custom https://pre-commit.ci/
but larger community still uses GH actions. And action by itself is pretty simple to break

good idea, I merge `continous-integration.yml` and added `pre-commit` as must have for publishing a build